pages/common/git-ls-tree.md

```diff
@@ -5,8 +5,12 @@
 
 - List the contents of the tree on a branch:
 
-`git ls-tree {{master}}`
+`git ls-tree {{branch_name}}`
 
-- List the contents of the tree on a commit, recurse into subtrees, and show only filenames:
+- List the contents of the tree on a commit and recurse into subtrees:
 
-`git ls-tree -r --name-only {{c39ab84d}}`
+`git ls-tree -r {{commit_hash}}`
+
+- List the contents of the tree on a commit and show only filenames:
+
+`git ls-tree --name-only {{HEAD~3}}`
```

Comment:
I would change `HEAD~3` to `HEAD~N` and note it shows the situation N commits ago or something like that.

Reply:
👍 

a2c3e613135cec75a31027ea1e7dc6535dd1c46a

Reply:
I would rather not put it as {{{HEAD}}~{{N}}. Maybe just HEAD~{{N}}?

/cc @waldyrious what would be the proper syntax here?

Reply:
@BeLi4L maybe wait for a consensus to emerge in these discussions before applying the changes, otherwise you may need to do a lot of back and forth 😅 

As for the syntax, I'm afraid we're making things too complex. The syntax for relative references is not something that this page is responsible for introducing. My suggestion is to simply include the `3` in the command description, and use `{{HEAD~3}}` as the token, rather than generalize with `N`. WDYT?

Reply:
Or maybe we could bypass this conundrum altogether by using another type of reference, like a tag (e.g. `{{v1.2.3}}`.

Reply:
`{{HEAD~3}}` simply seems too hard coded for an example. Maybe it could be given without braces.

Reply:
Or we could simply keep `{{commit_hash}}`.

People will probably try any ref anyway, what do you think?

Reply:
Well, since we already have an example using a branch name, and one using a commit hash, I think a different type of reference for this one would be useful. I get the reluctance regarding the potential complexity of using a relative reference, but I wonder if you guys have similar reservations about using a tag as I suggested above.

Reply:
All these solutions are OK for me, just tell me what you prefer ;)

Reply:
My preference is using a tag. Looking forward to @zdroid's thoughts.

Reply:
A `{{commit_hash}}` sounds the best to me out of all proposed solutions. I say go with it.

Reply:
Are you OK with d3cccfffc26bea964548c255666f92162627ac73?

Reply:
@zdroid I'd like to know what you dislike about using a tag, in light of my comments above.

Reply:
Isn't the term commit-ish used in the man pages?

Reply:
Any news on this? I can use `commit-ish` if you're all OK with that

Reply:
The current version is fine with me. Maybe @sbrl wants to use "commit-ish" instead, both work for me.

Reply:
I don't mind. Either work for me.